File: fern/snippets/tool-calling/versioning/python/manage-versions.py
from composio import Composio

composio = Composio(
    api_key="YOUR_API_KEY",
    toolkit_versions={"github": "20250116_00"}
)

# List available versions for a toolkit
versions = composio.tools.get_toolkit_versions("github")
for version in versions[:3]:
    print(f"Version: {version['version']} - Status: {version['status']}")

# Fetch tools and inspect their versions
tools = composio.tools.get("user-123", toolkits=["github"])
for tool in tools[:3]:
    print(f"{tool.slug}: v{tool.toolkit_version}")

File: fern/snippets/tool-calling/versioning/typescript/manage-versions.ts
import { Composio } from "@composio/core";

const composio = new Composio({
    apiKey: "YOUR_API_KEY",
    toolkitVersions: { github: "20250116_00" }
});

// List available versions for a toolkit
const versions = await composio.tools.getToolkitVersions("github");
versions.slice(0, 3).forEach(version => {
    console.log(`Version: ${version.version} - Status: ${version.status}`);
});

// Fetch tools and inspect their versions
const tools = await composio.tools.get("user-123", {
    toolkits: ["github"]
});
tools.slice(0, 3).forEach(tool => {
    console.log(`${tool.slug}: v${tool.toolkitVersion}`);
});